src/track/tracktiminginfo.cpp

```diff
@@ -0,0 +1,79 @@
+#include "track/tracktiminginfo.h"
+
+TrackTimingInfo::TrackTimingInfo(TrackPointer pTrack) :
+        m_pElapsedTimer(new TrackTimers::ElapsedTimerQt()),
+        m_pTimer(new TrackTimers::GUITickTimer()),
+        m_pTrackPtr(pTrack), 
+        m_playedMs(0),
+        m_isTrackScrobbable(false)
+{
+    connect(m_pTimer.get(),SIGNAL(timeout()),
+            this,SLOT(slotCheckIfScrobbable()));
+    m_pElapsedTimer->invalidate();
+}
+
+void TrackTimingInfo::pausePlayedTime() {    
+    if (m_pElapsedTimer->isValid()) {
+        m_playedMs += m_pElapsedTimer->elapsed();
+        m_pElapsedTimer->invalidate();
+    }        
+}
+
+void TrackTimingInfo::resumePlayedTime() {    
+    if (!m_pElapsedTimer->isValid()) {        
+        m_pElapsedTimer->start();
```

Comment:
What happens if the timer is already running from a different player with the same track?

Reply:
Nothing because then elapsed timer would be valid. But now that you mention I should probably check it beforehand just in case.